src/main/scala/latis/input/HapiAdapter.scala

```diff
@@ -4,40 +4,103 @@ import latis.model._
 import latis.util._
 import java.net.URI
 import latis.data.SampledFunction
-import latis.data.HapiFunction
+import latis.ops.Operation
+import latis.ops.Selection
+import latis.ops.Projection
 
 /**
  * Adapts a HAPI service as a source of data.
  */
-class HapiAdapter(model: DataType, config: HapiAdapter.Config) extends Adapter {
+abstract class HapiAdapter(model: DataType, config: HapiAdapter.Config) extends Adapter {
   
-  def apply(uri: URI): SampledFunction = 
-    HapiFunction (
-      model,
-      TextAdapter(model, config),
-      uri,
-      config.query
-    )
+  /**
+   * Defines the Adapter that will be used to parse the
+   * results from the HAPI service call
+   */
+  def parsingAdapter: Adapter
+  
+  /**
+   * Defines the requested data format.
+   * This must be consistent with the parsingAdapter.
+   */
+  def datasetFormat: String
+
+    
+
+  override def getData(baseUri: URI, ops: Seq[Operation] = Seq.empty): SampledFunction = {
+    val query = buildQuery(ops)
+    val uri = new URI(baseUri.toString + "data?" + query) //TODO: make sure baseURI has separator
+    parsingAdapter.getData(uri)
+  }
+    
+  override def canHandleOperation(op: Operation): Boolean = op match {
+    case Selection("time",_,_) => true
+    //case p: Projection => true
+    case _ => false
+  }
+  
+  def buildQuery(ops: Seq[Operation]): String = {
+    // Define the format of time strings used by the HAPI API.
+    //val timeFormat = TimeFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+    
+    // Placeholder for min time
+    var startTime: Option[String] = None
+    
+    // Placeholder for max time
+    var endTime:   Option[String] = None
```

Comment:
These variables don't appear to be used anywhere.

Reply:
These will be used to support the TODOs, to which I should probably do.